include/envoy/formatter/substitution_formatter.h

```diff
@@ -78,5 +79,34 @@ class FormatterProvider {
 
 using FormatterProviderPtr = std::unique_ptr<FormatterProvider>;
 
+class CommandParser {
+public:
+  virtual ~CommandParser() = default;
+
+  virtual FormatterProviderPtr parse(const std::string& token) const PURE;
```

Comment:
Please add comments for the class above and comments for this function. I had to read the code below to figure out how a format extension works. Please be detailed.